game-core/src/main/java/games/strategy/triplea/ui/PoliticalStateOverview.java

```diff
@@ -97,50 +109,38 @@ private void drawPoliticsUi() {
               0,
               0));
     }
+
     // draw cells
-    int x = 1;
-    int y = 2;
+    x = 1;
+    y = 2;
     for (final PlayerId verticalPlayer : data.getPlayerList()) {
       for (final PlayerId horizontalPlayer : data.getPlayerList()) {
-        if (horizontalPlayer.equals(verticalPlayer)) {
-          this.add(
-              new JLabel(PoliticalStateOverview.LABEL_SELF),
-              new GridBagConstraints(
-                  x++,
-                  y,
-                  1,
-                  1,
-                  1.0,
-                  1.0,
-                  GridBagConstraints.CENTER,
-                  GridBagConstraints.NONE,
-                  insets,
-                  0,
-                  0));
-        } else {
-          this.add(
-              getRelationshipLabel(verticalPlayer, horizontalPlayer),
-              new GridBagConstraints(
-                  x++,
-                  y,
-                  1,
-                  1,
-                  1.0,
-                  1.0,
-                  GridBagConstraints.CENTER,
-                  GridBagConstraints.BOTH,
-                  insets,
-                  0,
-                  0));
-        }
+        addCell(getRelationshipLabel(verticalPlayer, horizontalPlayer), insets, x++, y);
       }
       y = y + 2;
       x = 1;
     }
+
+    // Add horizontal struts for all the columns to make them have the same width.
+    for (int i = 0; i < data.getPlayerList().size(); i++) {
+      addCell(Box.createHorizontalStrut(maxColumnWidth), insets, i + 1, y);
+    }
+  }
+
+  protected JPanel wrapInJPanel(final JComponent component, final Color background) {
+    final JPanel p = new JPanel();
+    p.add(component);
+    if (background != null) {
+      p.setBackground(background);
+    }
+    return p;
   }
 
   /** Gets a label showing the colored relationshipName between these two players. */
   private JPanel getRelationshipLabel(final PlayerId player1, final PlayerId player2) {
+    if (player1.equals(player2)) {
+      return wrapInJPanel(new JLabel(PoliticalStateOverview.LABEL_SELF), null);
```

Comment:
code nit: try to avoid passing null values. https://github.com/triplea-game/triplea/wiki/Code-Conventions#null-handling

I think the helper method is perhaps not as needed as it could be.

We've a 'builder' helper library to fix some of the swing API problems, using that this is relatively straight forward:
```
new JPanelBuilder().add(new JLabel(PoliticalStateOverview.LABEL_SELF)).build();
```

We do not have a background property on `JPanelBuilder` it would be good to add, without doing that work, the other variant of the method call is pretty straight forward as well (line 168):

```
JPanel panel = new JPanelBuilder().addnew JLabel(relType.getName()).build();
panel.setBackgroundColor(relColor);
return panel;
```

Part of the motivation is to avoid different variable states, where 'null' background color means that parameter is unused, raises questions like "when is this null?" when is it not? What happens if it is. We also try to annotate nullable variables/values with "@Nullable" so that IDE can help detect NPE's for us, but first it's better to avoid null.